lib/pbench/server/api/resources/query_apis/metadata_index/__init__.py

```diff
@@ -91,13 +91,13 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
             self.logger.error(f"Indices from metadata table not found {e!r}")
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
-        index_keys = [key for key in index_map if root_index_name in key]
-        if len(index_keys) == 0:
-            self.logger.debug(
-                f"Found no indices matching the prefix {root_index_name}"
-                f"for a dataset {dataset!r}"
+        if index_map is None:
+            self.logger.error(
+                f"server.index-map not found in Metadata for a dataset {dataset!r}"
             )
-            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
```

Comment:
Should we be triggering on all "false-y" values of `index_map`?  (I.e., could `index_map` possibly be an empty list, rather than none?...in which case we probably want to respond the same way....)

Reply:
OK, nevermind, maybe?  It looks like the caller is coded to catch this case and report it as not-found rather than internal error.

Reply:
Nevermind...it looks like you guys discussed this above and I just missed it.